care/facility/api/serializers/facility.py

```diff
@@ -64,22 +68,30 @@ def validate_phone_number(self, value):
         return str(value).strip().replace("  ", " ")
 
     def create(self, validated_data):
-        capacities = validated_data.pop('facilitycapacity_set')
+        capacities = validated_data.pop("facilitycapacity_set")
         facility = Facility.objects.filter(
-            **{"name__iexact": validated_data['name'], "district": validated_data['district']}
+            **{
+                "name__iexact": validated_data["name"],
+                "district": validated_data["district"],
+            }
         ).first()
 
+        user = self.context["user"]
         if not facility:
+            validated_data["created_by"] = user
             facility = Facility.objects.create(**validated_data)
         else:
+            if facility.created_by != user and not user.is_superuser:
+                raise PermissionDenied(
+                    f"{facility.name}, {DISTRICT_CHOICES[facility.district][1]} is owned by another user"
```

Comment:
Can we reuse `__str__` here?

Reply:
https://github.com/coronasafe/care/pull/93/commits/8c2f73be03285ff242a9ccff7bd500cac4826655